ballerina/tests/csv_io.bal

```diff
@@ -99,6 +120,139 @@ isolated function testReadCsv() returns Error? {
     check csvChannel.close();
 }
 
+@test:Config {}
+isolated function testReadCsvRecord() returns Error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5.csv";
+    string[][] csvContent = check fileReadCsv(filePath);
+    test:assertEquals(csvContent[0][0], "User1");
+    test:assertEquals(csvContent[0][2], " 10000.50");
+    Employee[] employee = check fileReadCsv(filePath);
+    test:assertEquals(employee[0].id, "User1");
+    test:assertEquals(employee[0].salary, 10000.50f);
+}
+
+@test:Config {}
+function testReadFileCsvAsStreamUsingResourceFile() returns error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5.csv";
+    string[][] expected = [["User1", "WSO2", "10000.50"], ["User2", "WSO2", "20000.50"], ["User3", "WSO2", "30000.00"]];
+    stream<string[], Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(string[] val) {
+        foreach int j in 0 ... val.length()-1 {
+            test:assertEquals(val[j].trim(), expected[i][j]);
+        }
+        i += 1;
+    });
+    test:assertEquals(i, 3);
+}
+
+@test:Config {}
+function testReadFileCsvAsStreamUsingResourceFileRecord() returns error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5b.csv";
+    int[] expected = [10000,20000,30000];
+    stream<Employee4, Error?> result = check fileReadCsvAsStream(filePath); 
+    int i = 0;
+    check result.forEach(function(Employee4 val) {
+        test:assertEquals(val.salary, expected[i]);
+        i = i + 1;
+    });
+    test:assertEquals(i, 3);
+}
+
+@test:Config {}
+function testReadFileCsvAsStreamUsingResourceFileOpenRecord() returns error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5b.csv";
+    int[] expected = [10000,20000,30000];
+    stream<record {string id;string name;int salary;}, Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(record {string id; string name; int salary;} val) {
+        test:assertEquals(val.salary, expected[i]);
+        i = i + 1;
+    });
+    test:assertEquals(i, 3);
+}
+
+@test:Config {}
+isolated function testWriteRecordCsv() returns Error? {
+    Employee E = {
+        id: "1",
+        name: "Foo1",
+        salary: 100000.0f
+    };
+    Employee B = {
+        id: "2",
+        name: "Foo2",
+        salary: 300000.0f
+    };
+    string filePath = TEMP_DIR + "recordsDefault_records.csv";
+    Employee[] content1 = [E, B];
+    test:assertEquals(check fileWriteCsv(filePath, content1), ());
+}
+
+@test:Config {dependsOn: [testWriteRecordCsv]}
+function testWriteRecordCsvRead() returns error? {
```

Comment:
Is the name of the test case correct since the `fileReadCsvAsStream()` API is being tested?